cmd/tailscale/cli/up.go

```diff
@@ -105,7 +105,7 @@ func newUpFlagSet(goos string, upArgs *upArgsT) *flag.FlagSet {
 	upf.StringVar(&upArgs.advertiseRoutes, "advertise-routes", "", "routes to advertise to other nodes (comma-separated, e.g. \"10.0.0.0/8,192.168.0.0/24\") or empty string to not advertise routes")
 	upf.BoolVar(&upArgs.advertiseDefaultRoute, "advertise-exit-node", false, "offer to be an exit node for internet traffic for the tailnet")
 	if safesocket.GOOSUsesPeerCreds(goos) {
-		upf.StringVar(&upArgs.opUser, "operator", "", "Unix username to allow to operate on tailscaled without sudo")
+		upf.StringVar(&upArgs.opUser, "operator", os.Getenv("USER"), "Unix username to allow to operate on tailscaled without sudo")
```

Comment:
this looks like a change in behavior?

Reply:
this + additions on line 809-812 replicate the removed function `applyImplicitPrefs()`

Reply:
We can't make this change. Like Maisem says, this is a behavior change. We can't opt people in to this feature.

Reply:
when I traced the behavior of the flags through the setup we initialize the default env value to be user [here](https://github.com/tailscale/tailscale/blob/4005134263db39b5a9e0f13f6db0e260f871a5c7/cmd/tailscale/cli/up.go#L463) and then we had an implicit check in the `applyImplicitPrefs()` that replaced that value with any non-empty string value set in a flag. 

This, to me, is the exact same functionality as explicitly setting the default value to user and removing the check for non-empty string. 

If this is an incorrect understanding of functionality, please help me understand? I might need an additional conversation on a meet or slack hangout to really understand the difference. 

additionally, what defines behavior we can and can't change?

Reply:
[the docs](https://tailscale.com/kb/1080/cli/) say that the default is `root`. So if you could help me understand what I am not understanding by suggesting this change

Reply:
This value appears in the `up --help` text. I don't want to advertise any default for that flag, as by default there's no operator user.

That is, I don't like how it now does this:
```
bradfitz@tsdev:~/src/tailscale.com$ go run ./cmd/tailscale up --help 2>&1 | grep -A 1 operator
  --operator string
        Unix username to allow to operate on tailscaled without sudo (default bradfitz)

bradfitz@tsdev:~/src/tailscale.com$ go run --exec=sudo ./cmd/tailscale up --help 2>&1 | grep -A 1 operator
  --operator string
        Unix username to allow to operate on tailscaled without sudo (default root)
```

Both are misleading: the first makes it sound like the default is that `bradfitz` can operate `tailscaled` without sudo. (not true, but users might think that if they were the ones who installed it or something, guessing that we might've noticed that).  And the "root can operate tailscaled without sudo" part is also misleading, as root can always do anything without sudo.

Also, the fact that you had to add `wantOperatorChange: true` to some of the unrelated test cases is a bit of a red flag that there are behavior changes.

Reply:
> I don't like how it now does this
This to me should like we need a documentation change to match existing functionality. I am happy to work on that if it is needed. 
>  I don't want to advertise any default for that flag 
After the investigation and work put into the is #3808 I find that the only way to achieve the desired reset behavior without explicitly calling the `--reset` flag is to change functionality. If that is bigger than the scope of this issue I wonder if it is worth addressing.